src/test/java/com/databasepreservation/Roundtrip.java

```diff
@@ -0,0 +1,284 @@
+		this.environment_variables_source = environment_variables_source;
+		this.environment_variables_target = environment_variables_target;
+
+		processSTDERR = File.createTempFile("processSTDERR", ".tmp");
+		processSTDOUT = File.createTempFile("processSTDOUT", ".tmp");
+		processSTDERR.deleteOnExit();
+		processSTDOUT.deleteOnExit();
+	}
+
+	public Roundtrip(String setup_command, String teardown_command, String populate_command,
+			String dump_source_command, String dump_target_command,
+			String[] forward_conversion_arguments, String[] backward_conversion_arguments) throws IOException{
+		this.setup_command = setup_command;
+		this.populate_command = populate_command;
+		this.teardown_command = teardown_command;
+		this.dump_source_command = dump_source_command;
+		this.dump_target_command = dump_target_command;
+		this.forward_conversion_arguments = forward_conversion_arguments;
+		this.backward_conversion_arguments = backward_conversion_arguments;
+		this.environment_variables_source = new HashMap<String, String>();
+		this.environment_variables_target = new HashMap<String, String>();
+
+		processSTDERR = File.createTempFile("processSTDERR", ".tmp");
+		processSTDOUT = File.createTempFile("processSTDOUT", ".tmp");
+		processSTDERR.deleteOnExit();
+		processSTDOUT.deleteOnExit();
+	}
+
+	public boolean testTypeAndValue(String template, String... args) throws IOException, InterruptedException{
+		File populate_file = File.createTempFile("roundtrip_populate", ".sql");
+
+		FileWriter fw = new FileWriter(populate_file);
+		fw.write(String.format(template, args));
+		fw.write("\n");
+		fw.close();
+
+		setup();
+		boolean result = roundtrip(populate_file);
+		teardown();
+		if( populate_file.exists() )
+			populate_file.delete();
+		return result;
+	}
+
+	public boolean testFile(File populate_file) throws IOException, InterruptedException{
+		assert setup() == 0 : "Roundtrip setup exit status was not 0";
+		boolean result = roundtrip(populate_file);
+		assert teardown() == 0 : "Roundtrip teardown exit status was not 0";
+		return result;
+	}
+
+	/**
+	 * Runs a roundtrip test
+	 *
+	 * @param populate_file File with queries to populate the database
+	 * @return A diff string if the dumps differ, null otherwise
+	 * @throws IOException
+	 * @throws InterruptedException
+	 */
+	private boolean roundtrip(File populate_file) throws IOException, InterruptedException{
+		ProcessBuilder sql = new ProcessBuilder("bash", "-c", populate_command);
+		sql.redirectOutput(processSTDOUT);
+		sql.redirectError(processSTDERR);
+		sql.redirectInput(populate_file);
+		for(Entry<String, String> entry : environment_variables_source.entrySet()) {
+		    sql.environment().put(entry.getKey(), entry.getValue());
+		}
+		Process p = sql.start();
+		printTmpFileOnError(processSTDERR, p.waitFor());
+		printTmpFileOnError(processSTDOUT, p.waitFor());
+
+
+		Path dumpsDir = Files.createTempDirectory("dpttest_dumps");
```

Comment:
Do not mix NIO2 with IO.

Substitute all File with Path and use Paths and Files utils
Or refrain from using NIO2 (Path and Files)